test/com/google/javascript/jscomp/integration/AdvancedCoverageInstrumentationPassIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+package com.google.javascript.jscomp.integration;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static com.google.javascript.rhino.testing.NodeSubject.assertNode;
+
+import com.google.common.base.Joiner;
+import com.google.javascript.jscomp.BlackHoleErrorManager;
+import com.google.javascript.jscomp.Compiler;
+import com.google.javascript.jscomp.CompilerOptions;
+import com.google.javascript.jscomp.CompilerOptions.InstrumentOption;
+import com.google.javascript.jscomp.CompilerOptions.LanguageMode;
+import com.google.javascript.jscomp.SourceFile;
+import com.google.javascript.jscomp.testing.NoninjectingCompiler;
+import com.google.javascript.rhino.Node;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class AdvancedCoverageInstrumentationPassIntegrationTest extends IntegrationTestCase {
+
+
+  private final String instrumentCodeSource = lines("goog.module('instrument.code'); ",
+      "class InstrumentCode {",
+      "instrumentCode(a,b) {} ",
+      "}",
+      "const instrumentCodeInstance = new InstrumentCode();",
+      "exports = {instrumentCodeInstance};");
+
+  private final String instrumentCodeExpected = lines(
+      "var module$exports$instrument$code = {}",
+      "var module$contents$instrument$code_InstrumentCode = function() {};",
+      "module$contents$instrument$code_InstrumentCode.prototype.instrumentCode = function(a, b) {};",
+      "module$exports$instrument$code.instrumentCodeInstance = ",
+      "new module$contents$instrument$code_InstrumentCode;");
+
+  @Test
+  public void testFunctionInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source =
+        lines("function foo() { ",
+            "console.log('Hello'); ",
+            "}");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines(
+            "function foo() { ",
+            "module$exports$instrument$code.instrumentCodeInstance.instrumentCode(\"C\", 1);",
+            "console.log('Hello'); ",
+            "}");
+
+    String[] expectedArr = {instrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  @Test
+  public void testNoFunctionInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source =
+        lines("var global = 23; console.log(global);");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines("var global = 23; console.log(global);");
+
+    String[] expectedArr = {instrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  protected void test(CompilerOptions options,
```

Comment:
Is it necessary to override this method?

Reply:
This one is not so I reverted it but the below Compile method does need to be overridden.